src/dune_rules/odoc.ml

```diff
@@ -353,6 +351,10 @@ let setup_html sctx (odoc_file : odoc_artefact) =
       "html-generate" ~flags_for:None
       [ A "-o"
       ; Path (Path.build (Paths.html_root ctx))
+      ; A "--support-uri"
+      ; A "_odoc_support"
```

Comment:
Also, why is this not passed as a path? E.g. `Path (odoc_support ctx)`

Reply:
The rules are all context specific, so I don't see why not. When generating the command, dune will do `Path.reach` from the path from which the command is ran. So it should be a relative path in the end anyway.

Reply:
Hrm, actually it feels a bit misleading to me, as `Path.reach` isn't the right thing to do, even if we arrange for it to produce the correct results in this instance.

Reply:
Why isn't it the right thing to do?

Reply:
because it _isn't_ a path that's relative to the path from which the command is run - so dune and odoc are talking about different things if we do that. Dune can't make any inference about what's found at that path.

Reply:
I'd be more comfortable making it explicit - ie, `Path.reach Paths.html_dir Paths.support_dir` or something maybe?

Reply:
Looking at the docs I see:

```
       --support-uri=URI
           Where to look for support files (e.g. `URI/highlite.pack.js').
           Relative URIs are resolved using `--output-dir' as a target.
```

And paths are certainly valid URI's.

> although even that might not be right - I wouldn't want a ./ to get put on the front, for example.

Does that actually break it? I would hope not.

Reply:
Acutally, I suppose it should be `A (Path.reach support_dir ~from:index_html)` to be fully correct.

Reply:
> > although even that might not be right - I wouldn't want a ./ to get put on the front, for example.
> 
> Does that actually break it? I would hope not.

Ah probably not, and if so, that's an odoc bug :-) I've gone for this approach.